File: server/project/a/a/Models/Purchase.cs
namespace NET.Models
{
    public class Purchase
    {
         public int Id { get; set; }
         public int UserId { get; set; }
        public User User { get; set; }
         public int PresentId { get; set; }
        public Present Present { get; set; }
         public int NumOfTickets { get; set; }
        public bool IsDraft { get; set; } = false;
        public string Created_At { get; set; } = string.Empty;
        //public ICollection<Present> presents { get; set; } = new List<Present>();
    }
}

File: client/src/app/models/Purchase.ts
import { Present } from "./Present";

 export class Purchase
{
      id?:number;
      userId?:number;
      presentId!:number;
      Created_At?:string;
      present?:Present;
      userName?:string
      user?:User;
      numOfTickets!:number;
      isDraft:boolean=true;
}
    export class User
{
      id?:number;
      firstName!:string;
      lastName!:string;
      email!:string;
      address!:string;
      password!:string;
      phone!:string;
      created_At?:Date;
}

export class Category
{
    id?:number;
    name!:string;
}
export class Donor
{
    id?:number;
    name!:string;
    email!:string;
    created_At!:Date;
    presents:Present[]=[];
}
export class LotteryResult
{
    id?:number;
    presentId?:number;
    present?:Present;
    userId?:number;
    user?:User;
    createdAt?:Date;
}